landoapi/stacks.py

```diff
@@ -135,14 +137,13 @@ def request_extended_revision_data(
 
 class RevisionStack:
     def __init__(self, nodes: set[str], edges: set[tuple[str, str]]):
-        self.nodes = nodes
-        self.edges = edges
-
-        self.children = {phid: set() for phid in self.nodes}
-        self.parents = {phid: set() for phid in self.nodes}
-        for child, parent in self.edges:
-            self.children[parent].add(child)
-            self.parents[child].add(parent)
+        self.graph = nx.DiGraph(
+            # Reverse the order of the nodes in the edges set as `networkx`
+            # represents `a -> b` as `(a, b)` but Lando uses `(b, a)`.
+            (successor, predecessor)
+            for predecessor, successor in edges
+        )
```

Comment:
Haven't looked at the entire diff yet, but there's a `reverse` method that you may be able to use. See my changes in [revision_worker.py](https://github.com/mozilla-conduit/lando-api/pull/224/files#diff-b666a8fe8b2e34d21a8bf41d80705719f945165e15483472bc0d625707e3c8e1R58) for an example.

Reply:
We don't need to reverse the order of the edges list, but the order of the node relationships (ie the elements of the list) from `(dest, source)` to `(source, dest)`.

Touching some of the edge/graph related code in Lando/moz-phab the last little while has made me think we should use a more concrete type like a `namedtuple` to represent the edges, that way the order doesn't actually matter and we can just do `edge.source` and `edge.dest`. That's out of scope for this change of course. :)